eliot/_asyncio.py

```diff
@@ -31,3 +33,11 @@ def get_stack(self):
         if task not in self._per_task:
             self._per_task[task] = []
         return self._per_task[task]
+
+
+class AsyncioExecutionContext(_ExecutionContext):
+    """ExecutionContext that supports asyncio sub-contexts."""
+
+    def __init__(self):
+        _ExecutionContext.__init__(self)
+        self.get_sub_context = AsyncioSubContext().get_stack
```

Comment:
It looks like this is the fix for the multithreaded asyncio scenario described in the linked issue?  Previously the global `_ExecutionContext` instance was consulted for the current action and would only have the asyncio-aware `get_sub_context` implementation in the same thread as `use_asyncio_context` was called in.  Now ... I don't know.  How does it work now?  The execution context is looked up in the global `_ECOwner` instance.  If you called `use_asyncio_context` then you get back a global `AsyncioExecutionContext`.  Ah, yes, and that has a `get_sub_context` method that is thread-local and knows how to find the per-asyncio Task action stack.

Okay.